cadence/contracts/ISPOManager.cdc

```diff
@@ -552,33 +557,41 @@ pub contract ISPOManager {
     pub let ispoClientStoragePath: StoragePath
 
     pub struct ISPOClientInfo {
+        access(self) let ispoClientId: UInt64
         access(self) let ispoId: UInt64
         access(self) let delegatedFlowBalance: UFix64
         access(self) let rewardTokenBalance: UFix64
         access(self) let createdAt: UFix64
+        access(self) let hasDelegation: Bool
 
         init(
+            ispoClientId: UInt64,
             ispoId: UInt64,
             delegatedFlowBalance: UFix64,
             rewardTokenBalance: UFix64,
             createdAt: UFix64,
+            hasDelegation: Bool,
         ) {
+            self.ispoClientId = ispoClientId
             self.ispoId = ispoId
             self.delegatedFlowBalance = delegatedFlowBalance
             self.rewardTokenBalance = rewardTokenBalance
             self.createdAt = createdAt
+            self.hasDelegation = hasDelegation
```

Comment:
this is just for the UI to know whether to allow withdrawing and doesn't really help the contract be any more consistent inside after withdrawing